src/modules/storage/LedgerStorage.ts

```diff
@@ -2188,4 +2198,95 @@ export class LedgerStorage extends Storages
             `DROP DATABASE ${database}`;
        return this.run(sql, []);;
     }
+
+    /**
+     * Get Balance, freeze amount, tsx_count for a given address
+     * @returns Returns the Promise with requested data
+     * and if an error occurs the `.catch` is called with an error. 
+     */
+     public getBOAHolder(address:string): Promise<any>{
+        return new Promise<any>((resolve, reject) =>
+        {
+            (async () =>
+            { 
+                let freeze_amount = JSBI.BigInt(0);
+                let txs_count = 0;
+                let balance = JSBI.BigInt(0);
+                let balance_sql  =       `SELECT
+                                              SUM(amount) as amount
+                                              FROM
+                                              utxos 
+                                              WHERE address = ?;`
+                let tx_count_sql  =       `SELECT
+                                              count(address) as tx_count
+                                              FROM
+                                              tx_outputs
+                                              where address = ?;`  
+                 let freeze_sql   =        `SELECT 
+                                               SUM(amount) as freeze_amount
+                                               FROM 
+                                               utxos
+                                               WHERE 
+                                               type = 1 and address = ?;`
+
+                this.query(balance_sql,[address!.toString()])
+                .then(async(amount:any)=>{
+                    balance = amount[0].amount;
+                   return this.query(tx_count_sql,[address!.toString()]);
+                }).then((count: any)=>{
+                   txs_count = count[0].tx_count;
+                   return this.query(freeze_sql,[address!.toString()])
+                }).then((freeze: any)=>{
+                    freeze_amount = freeze[0].freeze_amount;                    
+                   let res = {
+                       balance,
+                       txs_count,
+                       freeze_amount,
+                       address,
+                       received_rewards:468768,
+                       precentage:'45%',
+                       value:'$8468'
```

Comment:
Why do you take the sum from raw data?
This can cause significant performance degradation.
Data should be queried in the `account` table.